crates/sui-framework/sources/sui.move

```diff
@@ -3,19 +3,22 @@
 
 /// Coin<SUI> is the token used to pay for gas in Sui
 module sui::sui {
-    use sui::balance::{Self, Supply};
-    use sui::coin;
+    use sui::tx_context::TxContext;
+    use sui::balance::Supply;
     use sui::transfer;
+    use sui::coin;
 
     friend sui::genesis;
 
     /// Name of the coin
     struct SUI has drop {}
 
-    /// Register the token to acquire its `TreasuryCap`.
+    /// Register the `SUI` Coin to acquire its `Supply`.
     /// This should be called only once during genesis creation.
-    public(friend) fun new(): Supply<SUI> {
-        balance::create_supply(SUI {})
+    public(friend) fun new(ctx: &mut TxContext): Supply<SUI> {
+        coin::treasury_into_supply(
+            coin::create_currency(SUI {}, 6, ctx)
```

Comment:
```suggestion
            coin::create_currency(SUI {}, 9, ctx)
```

The minimum unit is 10^-9 SUI, which we call a "mist" (i.e., 1 nano-SUI = 1 mist). I think we should also have a comment explaining this, perhaps on a `DECIMALS` constant so this information is easier to discover for someone reading the module?